clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -19681,6 +19681,21 @@ Value *CodeGenFunction::EmitHLSLBuiltinExpr(unsigned BuiltinID,
         getDotProductIntrinsic(CGM.getHLSLRuntime(), VecTy0->getElementType()),
         ArrayRef<Value *>{Op0, Op1}, nullptr, "hlsl.dot");
   }
+  case Builtin::BI__builtin_hlsl_dot2add: {
+    llvm::Triple::ArchType Arch = CGM.getTarget().getTriple().getArch();
+    if (Arch != llvm::Triple::dxil) {
+      llvm_unreachable("Intrinsic dot2add can be executed as a builtin only on dxil");
+    }
```

Comment:
IMO you  should use assert here. Its not that this is "unreachable" its that arriving here is a developer bug

```suggestion
    assert(Arch == llvm::Triple::dxil && "Intrinsic dot2add can be executed as a builtin only on dxil");
```

Reply:
also its worth noting that LLVM style says to drop the `{ }` for single line `if/for`

Reply:
This code will likely be temporary. I plan on deleting it after this pr merges and I land a patch for https://github.com/llvm/llvm-project/issues/132303.